drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -56,6 +56,22 @@ void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
     /* initialize device descriptor */
     dev->params = *params;
 #endif
+
+    do {
+        /* get an 8-byte unique ID to use as hardware address */
+        luid_get(dev->netdev.long_addr, IEEE802154_LONG_ADDRESS_LEN);
+
+        /* repeat until the address is marked as non-multicast and not globally
+         * unique. (luid_get() will manipulate the first byte on each call.
+         * we cannot just modify it, as this would risk collisions if multiple
+         * transceivers are connected on the board.)
+         */
+    }
+    while ((dev->netdev.long_addr[0] & 0x03) == 0x02);
```

Comment:
It was better before, now it makes completely no sense anymore.... first of all it wastes a lot of cycles + Now you are just checking if the bits at the first byte are set, but use it internally still in the order at hand (so the address is now used in the wrong order internally).

Reply:
OK. Let me explain the issue with just setting the bits of the first addresses.

Let's say the CPU ID is `0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`. In that case three consecutive calls to `luid_get()` with 8 bytes will return:

1. `0x01, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`
2. `0x02, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`
3. `0x03, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`

Those would now be changed to:

1. `0x02, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`
2. `0x02, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`
3. `0x02, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07`

So all three at86rf233 transceivers would gain the very same L2 address, which makes no sense.

Reply:
I fixed the first sentence in the reply above

Reply:
At this point, why not just swap the first two bytes you get from `luid_get()`…?

Reply:
> At this point, why not just swap the first two bytes you get from `luid_get()`…?

Yes, a much better idea!

Reply:
That's how it was originally...

Reply:
> That's how it was originally...

No. See: https://github.com/RIOT-OS/RIOT/blob/ea4b78654f45a33551d32ca205f35eac0aad4f56/drivers/at86rf2xx/at86rf2xx.c#L104-L107

Reply:
The current implementation changes the first byte returned by `luid_get()`. In context of the API of `luid_get()` every change of the returned locally unique ID breaks the guarantee of its uniqueness. In context of the current implementation, only the first byte must not be changed. The current code however does just this. Having two `at86rf233` transceivers on the same board with the current RIOT master will for some CPU IDs result in both transceivers having the same L2 address. That is one of the two bugs this PR tries to fix.

Reply:
This bug is also why one reset in https://github.com/RIOT-OS/RIOT/issues/9656 didn't break anything for the reporter, as that generated (due to that bug) the very same L2 address again. only after up to three resets a different L2 address is generated.

Reply:
Being honest: I still think that even with this PR applied the driver is still wrongly using `luid_get()`. It is totally insane to me to ask a unique 8 byte array, modify it, and still expect it to be unique. It only works now due to implementation details of `luid_get()`. Changing them would again break the L2 address generation.

If the standard use case of `luid` is to generate EUI64 and EUI48, maybe it is the time to add `luid_get_eui64()` and `luid_get_eui48()`? Those could enforce that the generated IDs are correctly marked as unicast and non-globally-unique and still unique for the boards they were generated. And this as part of the documentated guarantees of the API, not as a random internal implementation detail that might change. Also this would likely dedpulicate code.

Reply:
> If the standard use case of luid is to generate EUI64 and EUI48, maybe it is the time to add `luid_get_eui64()` and `luid_get_eui48()`?

Yes, please!
A quick `grep` shows that *all* invocations but one of `luid_get()` are to set a MAC Address.
And everywhere those two magical bits are flipped.

If this discussion has has shown something, than it is that those functions are direly needed.

Also mind that all (?) drivers currently use the first two bytes of the long address for the short address, but I don't know if this is necessary. (I doubt so)